connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitter.java

```diff
@@ -67,70 +69,53 @@ public void close(long timeoutMs) {
     public void schedule(final ConnectorTaskId id, final WorkerSourceTask workerTask) {
         synchronized (committers) {
             long commitIntervalMs = config.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
-            ScheduledFuture<?> commitFuture = commitExecutorService.schedule(new Runnable() {
+            ScheduledFuture<?> commitFuture = commitExecutorService.scheduleWithFixedDelay(new Runnable() {
                 @Override
                 public void run() {
                     commit(id, workerTask);
                 }
-            }, commitIntervalMs, TimeUnit.MILLISECONDS);
-            committers.put(id, new ScheduledCommitTask(commitFuture));
+            }, 0, commitIntervalMs, TimeUnit.MILLISECONDS);
+            committers.put(id, commitFuture);
```

Comment:
Oh now I see. There's an interdependency between `commit` and the map. Will revisit.